packages/vite/src/node/utils.ts

```diff
@@ -720,36 +720,44 @@ interface ImageCandidate {
   url: string
   descriptor: string
 }
-const escapedSpaceCharacters = /(?: |\\t|\\n|\\f|\\r)+/g
-const imageSetUrlRE = /^(?:[\w-]+\(.*?\)|'.*?'|".*?"|\S*)/
+
 function joinSrcset(ret: ImageCandidate[]) {
   return ret
     .map(({ url, descriptor }) => url + (descriptor ? ` ${descriptor}` : ''))
     .join(', ')
 }
 
-// NOTE: The returned `url` should perhaps be decoded so all handled URLs within Vite are consistently decoded.
-// However, this may also require a refactor for `cssReplacer` to accept decoded URLs instead.
-function splitSrcSetDescriptor(srcs: string): ImageCandidate[] {
-  return splitSrcSet(srcs)
-    .map((s) => {
-      const src = s.replace(escapedSpaceCharacters, ' ').trim()
-      const url = imageSetUrlRE.exec(src)?.[0] ?? ''
+const escapedSpaceCharacters = /(?: |\\t|\\n|\\f|\\r)+/g
 
-      return {
-        url,
-        descriptor: src.slice(url.length).trim(),
-      }
-    })
-    .filter(({ url }) => !!url)
+export function parseSrcset(string: string): ImageCandidate[] {
+  const matches = string
+    .trim()
+    .replace(escapedSpaceCharacters, ' ')
+    .replace(/\r?\n/, '')
+    .replace(/,\s+/, ', ')
+    .replaceAll(/\s+/g, ' ')
+    .matchAll(
+      /**
+       This regex represents a loose rule of an “image candidate string” and "image set options".
+
+       @see https://html.spec.whatwg.org/multipage/images.html#srcset-attribute
+       @see https://drafts.csswg.org/css-images-4/#image-set-notation
+       */
+      // eslint-disable-next-line regexp/no-super-linear-backtracking
+      /\s*(?<url>[\w-]+\([^)]*\)|"[^"]*"|'[^']*'|[^,]\S*[^,])\s*(?<descriptor>[^,]*(?:\stype\([^)]*\))?)\s*(?:,|$)/g,
```

Comment:
```suggestion
      /(?:^|\s)(?<url>[\w-]+\([^)]*\)|"[^"]*"|'[^']*'|[^,]\S*[^,])\s*(?:\s(?<descriptor>[^,\s]+(?:\stype\([^)]*\))?)\s*)?(?:,|$)/g,
```
I think this lint can be fixed by replacing with this regex. The diffs are:

- replaced the first `\s*` with `(?:^|\s)`
- wrap descriptor part with `(?:\s` + `)?` and swapped `[^,]*` inside with `[^,\s]+`: require a single `\s` if a descriptor exists

Reply:
Thank you. Added your suggestions, extracted a variable and extended the description in the comment.

Reply:
There is an ordering problem, `type` can also be before `1x`

Reply:
Should be fixed.